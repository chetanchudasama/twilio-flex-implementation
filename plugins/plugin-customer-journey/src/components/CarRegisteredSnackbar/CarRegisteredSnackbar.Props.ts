export interface CarRegisteredSnackbarProps {
  is247Cars: boolean;
}
