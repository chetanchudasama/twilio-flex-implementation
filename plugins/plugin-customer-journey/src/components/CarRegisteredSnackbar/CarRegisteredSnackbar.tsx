import React from "react";
import { CustomSnackbar } from "@common/components";

import { CarRegisteredSnackbarProps } from "./CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = ({
  is247Cars,
}) => {
  return (
    <>
      {is247Cars ? (
        <CustomSnackbar
          type="success"
          message="Great news! This car is on our car search with one of our approved dealers!"
        />
      ) : (
        <CustomSnackbar
          type="info"
          message="This car isn't in our car search."
        />
      )}
    </>
  );
};
