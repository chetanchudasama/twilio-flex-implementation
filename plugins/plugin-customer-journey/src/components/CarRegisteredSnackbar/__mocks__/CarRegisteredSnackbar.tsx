import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
