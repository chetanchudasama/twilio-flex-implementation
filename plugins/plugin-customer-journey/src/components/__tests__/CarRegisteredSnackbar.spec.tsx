import React from "react";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import { CarRegisteredSnackbar } from "../CarRegisteredSnackbar/CarRegisteredSnackbar";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar/CarRegisteredSnackbar.Props";

const props: CarRegisteredSnackbarProps = {
  is247Cars: true,
};

describe("CarRegisteredSnackbar", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<CarRegisteredSnackbar {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("when value of is247Cars prop is true, should render warning message accordingly", () => {
    const wrapper = mount(<CarRegisteredSnackbar {...props} />);

    expect(wrapper.html()).toContain(
      "Great news! This car is on our car search with one of our approved dealers!"
    );
  });

  it("when value of is247Cars prop is false, should render warning message accordingly", () => {
    props.is247Cars = false;
    const wrapper = mount(<CarRegisteredSnackbar {...props} />);

    expect(wrapper.html()).toContain("This car isn't in our car search.");
  });
});
