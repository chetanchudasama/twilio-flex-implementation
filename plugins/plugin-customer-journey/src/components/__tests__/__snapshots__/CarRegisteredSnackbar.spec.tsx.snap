// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarRegisteredSnackbar renders correctly on load 1`] = `
<div
  className="MuiGrid-item-15 MuiGrid-grid-xs-12-54 CustomSnackbarContent-root-1"
>
  <div
    className="MuiTypography-root-141 MuiTypography-body1-150 MuiPaper-root-114 MuiPaper-elevation6-122 MuiSnackbarContent-root-111 CustomSnackbarContent-snackbarContainer-2 CustomSnackbarContent-success-8"
    role="alertdialog"
  >
    <div
      className="MuiSnackbarContent-message-112"
    >
      <span
        className="CustomSnackbarContent-message-4"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-177 CustomSnackbarContent-icon-6 CustomSnackbarContent-iconVariant-5"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        Great news! This car is on our car search with one of our approved dealers!
      </span>
    </div>
  </div>
</div>
`;
